src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs

```diff
@@ -570,8 +570,11 @@ private LdapConnection GetConnection()
                 AuthType = AuthType.Basic
             };
 
-            connection.Bind();
+            // Set server protocol before bind; OpenLDAP servers default
+            // to LDAP v2, which we do not support, and will return LDAP_PROTOCOL_ERROR
             connection.SessionOptions.ProtocolVersion = 3;
```

Comment:
I wonder if we should default to version 3 when creating the LdapConnection object since that will be what you want more than 90% of the time. I saw other libraries do default to 3 as well, for example: [https://github.com/flamencist/ldap4net/blob/41a6397d6ad2c3cbaad65aa39b637dfc81ffac49/LdapForNet/ILdapConnection.cs#L13](https://github.com/flamencist/ldap4net/blob/41a6397d6ad2c3cbaad65aa39b637dfc81ffac49/LdapForNet/ILdapConnection.cs#L13)